tron/serialize/runstate/dynamodb_state_store.py

```diff
@@ -119,37 +112,37 @@ def save(self, key_value_pairs) -> None:
         except Exception as e:
             self.alert(str(e))
```

Comment:
since we're using batch_writer we're safe(ish?) with regard to partial saves: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb.html#DynamoDB.Table.batch_writer

> ... the batch writer will also automatically handle any unprocessed items and resend them as needed.

it's still possible batch_writer will fail even after retrying write operations so would be useful to catch this case somehow.

Reply:
of course the right choice here is to use transact_write_items instead of batch_writer: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb.html#DynamoDB.Client.transact_write_items

Reply:
transact_write_item is a good idea. But I also have to add a retry alogrithm to it. In the worst case, I think we have to lose some data. I can set up scheduled backup to recover though.